### PR TITLE
8306466: Open source more AWT Drag & Drop related tests

### PR DESCRIPTION
**Pull request by DamonGuy**
This PR open sources a few more AWT DnD related tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306466](https://bugs.openjdk.org/browse/JDK-8306466): Open source more AWT Drag &amp; Drop related tests

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13653/head:pull/13653` \
`$ git checkout pull/13653`

Update a local copy of the PR: \
`$ git checkout pull/13653` \
`$ git pull https://git.openjdk.org/jdk.git pull/13653/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13653`

View PR using the GUI difftool: \
`$ git pr show -t 13653`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13653.diff">https://git.openjdk.org/jdk/pull/13653.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13653#issuecomment-1522215492)

### CODE REVIEW

**Comment by prrace** [comment]
The 1st 4 of these tests do not need their own directories since they are single file tests. The extra folder can be dropped

**Comment by DamonGuy** [comment]
> The 1st 4 of these tests do not need their own directories since they are single file tests. The extra folder can be dropped

Sure, I removed the folders and placed the tests directly into the dnd folder.

File: test/jdk/java/awt/dnd/RejectDragDropActionTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            }
+
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        } finally {
+            if (frame != null) {
+                frame.dispose();
+            }
+        }
+
+        if (incorrectActionDetected) {
```
Comment (mrserb) [inline test/jdk/java/awt/dnd/RejectDragDropActionTest.java:114]: incorrectActionDetected is set on one thread an used on another, same for "frame" in other tests.

Comment (prrace) [reply test/jdk/java/awt/dnd/RejectDragDropActionTest.java:114]: meaning you are suggesting they need to be volatile ?

Comment (mrserb) [reply test/jdk/java/awt/dnd/RejectDragDropActionTest.java:114]: or changed to be used on the same thread, in this case EDT. It seems different tests use different approaches.

File: test/jdk/java/awt/dnd/SkipDropCompleteTest/SkipDropCompleteTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException, AWTException {
+        SkipDropCompleteTest skipDropCompleteTest = new SkipDropCompleteTest();
+        EventQueue.invokeAndWait(skipDropCompleteTest::init);
+        skipDropCompleteTest.start();
+    }
+
+    public void init() {
+        sourceFrame = new SourceFrame();
+        targetFrame  = new TargetFrame();
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/SkipDropCompleteTest/SkipDropCompleteTest.java:56]:
```suggestion
        targetFrame = new TargetFrame();
```

File: test/jdk/java/awt/dnd/SkipDropCompleteTest/SourceFrame.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    public void dragOver(DragSourceDragEvent dsde) { }
+
+    public void dragExit(DragSourceEvent dse) { }
+
+    public void dropActionChanged(DragSourceDragEvent dsde ) { }
+
+    public void dragDropEnd(DragSourceDropEvent dsde) { }
+
+    public void dragGestureRecognized(DragGestureEvent dge)  {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/SkipDropCompleteTest/SourceFrame.java:82]:
```suggestion
    public void dragGestureRecognized(DragGestureEvent dge) {
```

File: test/jdk/java/awt/dnd/SkipDropCompleteTest/TransferableObject.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                System.out.println("Exception " + e);
+        }
+    }
+
+    DataFlavor[] dfs;
+
+    public TransferableObject(Object data) {
+        super();
+        Vector v = new Vector();
+            if(data instanceof String) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/SkipDropCompleteTest/TransferableObject.java:52]:
```suggestion
            if (data instanceof String) {
```

File: test/jdk/java/awt/dnd/SkipDropCompleteTest/SkipDropCompleteTest.java (hunk 2 above)
Comment (DamonGuy) [reply test/jdk/java/awt/dnd/SkipDropCompleteTest/SkipDropCompleteTest.java]: Fixed

File: test/jdk/java/awt/dnd/SkipDropCompleteTest/TransferableObject.java (hunk 4 above)
Comment (DamonGuy) [reply test/jdk/java/awt/dnd/SkipDropCompleteTest/TransferableObject.java:52]: Fixed

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**